bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/api/AgsHash.scala

```diff
@@ -56,18 +56,17 @@ object AgsHash {
       c.cc.## +=: c.iq.## +=: c.sb.## +=: buf
     }
 
-    // Base Position
-    Option(ctx.getTargets).foreach { t =>
+    // Asterism
+    Option(ctx.getTargets).foreach { env =>
       val time = Some(new java.lang.Long(when)).asGeminiOpt
-      val base = t.getBase
-
-      def toData(coord: GemOption[java.lang.Double]): Int =
-        coord.asScalaOpt.map(_.doubleValue).##
-
-      val ra  = toData(base.getRaDegrees(time))
-      val dec = toData(base.getDecDegrees(time))
-
-      ra +=: dec +=: buf
+      def hashTarget(t: SPTarget): Unit = {
+        def toData(coord: GemOption[java.lang.Double]): Int =
+          coord.asScalaOpt.map(_.doubleValue).##
+        val ra  = toData(t.getRaDegrees(time))
+        val dec = toData(t.getDecDegrees(time))
+        ra +=: dec +=: buf
+      }
+      env.getAsterism.targets.foreach(hashTarget)
     }
```

Comment:
👍  GHOST will use a combination of guide fibers (surrounding the science targets) and P2.  The guide fiber state won't be controlled by AGS, I assume, so it boils down to finding a P2 star.  It seems to me that the positions of the individual stars will matter there since we'll want to not only find a P2 star that is reachable from the base position but also one that doesn't vignette the science targets.